br/pkg/lightning/importer/import.go

```diff
@@ -1674,6 +1722,74 @@ func (rc *Controller) importTables(ctx context.Context) (finalErr error) {
 	return nil
 }
 
+func (rc *Controller) dropAllIndexes(ctx context.Context) error {
+	for _, dbInfo := range rc.dbInfos {
+		for _, tblInfo := range dbInfo.Tables {
+			if err := rc.dropTableIndexes(ctx, tblInfo); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (rc *Controller) dropTableIndexes(ctx context.Context, tblInfo *checkpoints.TidbTableInfo) error {
+	tableName := common.UniqueTable(tblInfo.DB, tblInfo.Name)
+	logger := log.FromContext(ctx).With(zap.String("table", tableName))
+
+	var remainIndexes []*model.IndexInfo
+	cols := tblInfo.Core.Columns
+loop:
+	for _, idxInfo := range tblInfo.Core.Indices {
+		if idxInfo.State != model.StatePublic {
+			remainIndexes = append(remainIndexes, idxInfo)
+			continue
+		}
+		// Primary key is a cluster index.
+		if idxInfo.Primary && tblInfo.Core.HasClusteredIndex() {
```

Comment:
primary key can be non-clustered
https://docs.pingcap.com/zh/tidb/dev/clustered-indexes

Reply:
For non-clustered primary key, `HasClusteredIndex` should return false.

Reply:
so only clustered-primary key is retained, all other index is removed?

Reply:
yes